### PR TITLE
TOOLS-2557 Update eng for new Rust Makefiles.

### PR DESCRIPTION
**Pull request by travispaul**
Updated deps/eng and Makefile for new Rust tooling.

Any changes to rust code are non-functional (the result of `cargo fmt` and copyright updates)

### CODE REVIEW

**Review by danmcd — DISMISSED** [review]
Makefile seems sensible.  Please get a second on this one, however.

File: buckets-mdapi/src/main.rs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 // Copyright 2020 Joyent, Inc.
+// Copyright 2023 MNX Cloud, Inc.
```
Comment (bahamat) [inline buckets-mdapi/src/main.rs:2]: This file didn't change other than the copyright.

File: buckets-mdapi/src/object.rs [hunk 2]
```diff
@@ -1,14 +1,15 @@
 // Copyright 2020 Joyent, Inc.
+// Copyright 2023 MNX Cloud, Inc.
 
 use std::error::Error;
 use std::vec::Vec;
 
 use base64;
 use bytes::BytesMut;
+use postgres::types::{accepts, to_sql_checked};
```
Comment (bahamat) [inline buckets-mdapi/src/object.rs:9]: Why did this move?

File: buckets-mdapi/src/opts.rs [hunk 3]
```diff
@@ -1,6 +1,5 @@
-/*
- * Copyright 2019 Joyent, Inc.
- */
+// Copyright 2019 Joyent, Inc.
+// Copyright 2023 MNX Cloud, Inc.
```
Comment (bahamat) [inline buckets-mdapi/src/opts.rs:2]: So, in general we should be using `/* ... */` style comment for copyright and license header. Which also brings up that none of these files have the license header in them.

This should be fixed at some point, but I'm not going to make it an issue for *this* PR.

File: .gitignore [hunk 4]
```diff
@@ -19,3 +19,4 @@ config.toml
 /make_stamps
 /bits
 *.tar.gz
+/rust
```
Comment (bahamat) [inline .gitignore:22]: Missing cache dir

**Review by bahamat — CHANGES_REQUESTED** [review]

File: buckets-mdapi/src/object.rs (hunk 2 above)
Comment (travispaul) [reply buckets-mdapi/src/object.rs:9]: `cargo fmt` will sort `use` statements alphabetically

**Review by bahamat — APPROVED** [review]

**Pull request merged**